### ISSUE TITLE
Next run's "scheduled for" time doesn't match the "Schedules" time

### ISSUE THREAD

**Issue by mitchnielsen** [PrefectHQ/prefect#13591]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I refreshed the page and this issue still occurred.
- [X] I checked if this issue was specific to the browser I was using by testing with a different browser.

### Bug summary

The time mentioned under "Next run" doesn't match with the time mentioned under "Schedules":

This is with my timezone set to `America/Chicago` under Preferences.

### Reproduction

1. [Create a work pool](https://docs.prefect.io/latest/getting-started/quickstart/#step-4-choose-a-remote-infrastructure-location):

```shell
prefect work-pool create my-managed-pool --type prefect:managed
```

2. [Create a scheduled Deployment](https://docs.prefect.io/latest/getting-started/quickstart/#step-5-make-your-code-schedulable):

```python
# create_deployment.py
from prefect import flow

if __name__ == "__main__":
    flow.from_source(
        source="https://github.com/prefecthq/demos.git",
        entrypoint="my_gh_workflow.py:repo_info",
    ).deploy(
        name="my-first-deployment",
        work_pool_name="my-managed-pool",
        cron="0 1 * * *",
    )
```

```shell
python create_deployment.py
```

3. Observe the times listed on the Deployment page.

### Error

Note that the "Next Run" section reports 8:00 PM, while the "Schedules" section reports 1:00 PM:

![image](https://github.com/PrefectHQ/prefect/assets/7725815/ae7fba98-0447-49f6-ad27-5988f7bfc2e2)

### Browsers

- [X] Chrome
- [X] Firefox
- [ ] Safari
- [ ] Edge

### Additional context

I'm still new to Prefect so it's very possible that this is either known or requires a setting I haven't set yet :)

**Comment by mitchnielsen** [comment]
Yup! Kansas City, set to `America/Chicago` in Preferences.

> a cron string sans timezone anchors to UTC

Mm, makes sense - I don't see a way at the moment to pass a timezone in `flow.deploy()`, but yeah, maybe we can eventually get the UI to align them based on the timezone set in Preferences.

**Comment by WillRaphaelson** [comment]
Yes after discussion we agree we should just make the UI display the consistent timezone.
